### PR TITLE
accommodating for blank confidence #97

### LINKED ISSUE #97: merge method does not work when no confidence present
the merge method should allow the case that a data frame provides no confidence. We assume confidence 1 when there is no confidence given.

### PR DESCRIPTION
**Pull request by hrshdhgd**
Fixes #97

### CODE REVIEW

File: sssom/sssom_datamodel.py [hunk 1]
```diff
@@ -299,7 +299,9 @@ def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):
         if self.mapping_date is not None and not isinstance(self.mapping_date, str):
             self.mapping_date = str(self.mapping_date)
 
-        if self.confidence is not None and not isinstance(self.confidence, float):
+        if self.confidence == '' or self.confidence is None:
```
Comment (matentzn) [inline sssom/sssom_datamodel.py:302]: Unfortunately sssom_datamodel.py cannot be manually editted, the other comment..

Comment (matentzn) [inline sssom/sssom_datamodel.py:302]: Unfortunately you cant edit sssom/sssom_datamodel.py manually, since its generated, see other comments.

**Review by matentzn — CHANGES_REQUESTED** [review]

File: sssom/sssom_datamodel.py (hunk 1 above)
Comment (hrshdhgd) [reply sssom/sssom_datamodel.py:302]: rolling it back.Thanks!

**Comment by hrshdhgd** [comment]
This fails in `diff` as well. Do you think there is a way to autogenerate this logic in `sssom_datamodel.py` such that it is taken care of for all commands (even for the ones in the future) rather than individually doing it for every command (`merge`, `diff` etc.).

**Comment by matentzn** [comment]
I tried using LinkML default values - that _could_ do it, but seems extremely risky.. Like assuming _in general_ that confidence is 1.0.. The default confidence should be unknown..

cant we just add an if else to the merge() function? what would be the risk?

**Comment by hrshdhgd** [comment]
thoughts on the latest commit? https://github.com/mapping-commons/sssom-py/issues/97

**Comment by hrshdhgd** [comment]
Edited code to reflect our Slack conversation. Thoughts?

**Comment by hrshdhgd** [comment]
Tests pass now.

**Comment by matentzn** [comment]
Can you resolve the conflicts please (rebasing with master)?

**Comment by hrshdhgd** [comment]
Done

**Comment by hrshdhgd** [comment]
Done

**Comment by matentzn** [comment]
What is the state of this pull request - did you convince yourself that the merging still works well? If so, what is the reason for changing the tests exactly (see slack discussion?)

**Comment by hrshdhgd** [comment]
Yes, this works well. The test was changed because when I had introduced the blank confidence value, it was done in another branch. We rolled it back since it wasn't part of that PR but this PR. So I made that change here.

**Review by matentzn — APPROVED** [review]

**Comment by matentzn** [comment]
Thank you very much @hrshdhgd

**Pull request merged**